shipwright/base.py

```diff
@@ -15,6 +25,8 @@ def targets(self):
     def build(self, build_targets):
         targets = dependencies.eval(build_targets, self.targets())
         this_ref_str = self.source_control.this_ref_str()
+        if '-dirty-' in this_ref_str and not build_targets['dirty']:
```

Comment:
rather than checking if "-dirty-" is in this_ref_str, just pass allow_dirty into the GitSourceControll class